### PR TITLE
NFSE-5080-cnget-perfc

### PR DESCRIPTION
**Pull request by beckerg**
Signed-off-by: Greg Becker <gbecker@micron.com>

### CODE REVIEW

**Comment by beckerg** [comment]
@bhaveshvasandani  Looks like build on Fedora is failing because poetry is failing due to missing libs...

File: lib/cn/cn_tree.c [hunk 1]
```diff
@@ -1279,10 +1277,13 @@ cn_tree_lookup(
 
     pfx_hashing = kt->kt_len > tree->ct_pfx_len && node->tn_pfx_spill;
     first = true;
+    err = 0;
 
     rmlock_rlock(&tree->ct_lock, &lock);
     while (node) {
+		struct kvset_list_entry *le;
         bool yield = false;
+		u32 child;
```
Comment (gsramdasi) [inline lib/cn/cn_tree.c:1286]: The indentation looks off

File: lib/cn/cn_tree.c [hunk 2]
```diff
@@ -1364,22 +1365,17 @@ cn_tree_lookup(
         perfc_lat_record(pc, PERFC_LT_CNGET_PROBE_PFX, pc_start);
         kvset_wbti_free(wbti);
     } else {
-        uint pc_cidx_ra = (*res == NOT_FOUND) ? PERFC_RA_CNGET_MISS : PERFC_RA_CNGET_GET;
-
-        if (pc_start > 0) {
-            uint pc_cidx_lt = (*res == NOT_FOUND) ? PERFC_LT_CNGET_MISS : PERFC_LT_CNGET_GET;
+		if (pc_start > 0) {
```
Comment (gsramdasi) [inline lib/cn/cn_tree.c:1368]: indentation

File: lib/cn/cn_perfc.c [hunk 3]
```diff
@@ -18,26 +19,28 @@
 /* clang-format off */
 
 struct perfc_name cn_perfc_get[] _dt_section = {
-    NE(PERFC_RA_CNGET_GET,       2, "Count of cN lookup attempts",   "c_get(/s)"),
-    NE(PERFC_RA_CNGET_MISS,      2, "Count of cN lookup misses",     "c_mis(/s)"),
-    NE(PERFC_RA_CNGET_TOMB,      2, "Num lookups that found tombs",  "c_tmb(/s)"),
+    NE(PERFC_RA_CNGET_MISS,      2, "cN lookup miss rate",           "c_mis(/s)"),
```
Comment (gsramdasi) [inline lib/cn/cn_perfc.c:22]: The prefix 'r' probably means rate. What's 'c' for?

**Review by gsramdasi — APPROVED** [review]

File: lib/cn/cn_perfc.c (hunk 3 above)
Comment (gsramdasi) [reply lib/cn/cn_perfc.c:22]: Ah, probably count. But if they're all described as rate now maybe we can change them to r.

File: lib/cn/cn_tree.c (hunk 1 above)
Comment (beckerg) [reply lib/cn/cn_tree.c:1286]: Ahhh, my style selector was broken, was wondering why it was inserting tabs, thx!

File: lib/cn/cn_tree.c (hunk 2 above)
Comment (beckerg) [reply lib/cn/cn_tree.c:1368]: fixed

File: lib/cn/cn_perfc.c (hunk 3 above)
Comment (beckerg) [reply lib/cn/cn_perfc.c:22]: Right, I didn't want to change the existing ones for fear of breaking grafana.  But I want to come back with a different patch and fix them all, but need to coordinate with @twblamer to do that...

**Review by alexttx — APPROVED** [review]

**Pull request merged**